### PR TITLE
🐛 Nit: trim unncessary if check in templates/controllers/controller-test.go

### PR DESCRIPTION
**Pull request by LittleWat**
Thank you for your nice examples!! (linked from https://github.com/kubernetes-sigs/kubebuilder/discussions/2836#discussioncomment-3259637)

I just removed unnecessary if check

### CODE REVIEW

File: testdata/project-v3-with-deploy-image/controllers/busybox_controller_test.go [hunk 1]
```diff
@@ -85,9 +85,7 @@ var _ = Describe("Busybox controller", func() {
 				}
 
 				err = k8sClient.Create(ctx, busybox)
-				if err != nil {
-					Expect(err).To(Not(HaveOccurred()))
-				}
+				Expect(err).To(Not(HaveOccurred()))
```
Comment (camilamacedo86) [inline testdata/project-v3-with-deploy-image/controllers/busybox_controller_test.go:88]: Hi @LittleWat, 

Thank you for the contribution 🥇.

But this change cannot be done in the samples. the samples are generated automatically when we run `make generate`, see: https://github.com/kubernetes-sigs/kubebuilder/blob/master/test/testdata/generate.sh

Therefore, this change needs to be done in the plugin implementation: https://github.com/kubernetes-sigs/kubebuilder/blob/master/pkg/plugins/golang/deploy-image/v1alpha1/scaffolds/internal/templates/controllers/controller-test.go#L138-L140  

Could you please do that in the code implementation and run make generate to ensure that all samples will be updated according to, squash the commits and push it so we are able to get this one merged?

Comment (camilamacedo86) [reply testdata/project-v3-with-deploy-image/controllers/busybox_controller_test.go:88]: Also, please note that you need assign to the CLA. https://github.com/kubernetes-sigs/kubebuilder/pull/2838#issuecomment-1197606995

Comment (LittleWat) [reply testdata/project-v3-with-deploy-image/controllers/busybox_controller_test.go:88]: @camilamacedo86 Thank you for quick and detailed review! Oh, these test codes are generated, cool!

I edited the original template file and  did `make generate` and committed!
I also signed!

File: pkg/plugins/golang/deploy-image/v1alpha1/scaffolds/internal/templates/controllers/controller-test.go [hunk 2]
```diff
@@ -135,19 +135,13 @@ var _ = Describe("{{ .Resource.Kind }} controller", func() {
...
 
 			By("Checking if the custom resource was successfully created")
 			Eventually(func() error {
 				found := &{{ .Resource.ImportAlias }}.{{ .Resource.Kind }}{}
-				err = k8sClient.Get(ctx, typeNamespaceName, found)
-				if err != nil {
-					return err
-				}
-				return nil
+				return k8sClient.Get(ctx, typeNamespaceName, found)
```
Comment (LittleWat) [inline pkg/plugins/golang/deploy-image/v1alpha1/scaffolds/internal/templates/controllers/controller-test.go:144]: this is the same as 
https://github.com/openshift/operator-framework-olm/blob/3e1dc5ce2547e283949bef2f2cd992cab8ffd730/staging/operator-lifecycle-manager/pkg/controller/operators/openshift/clusteroperator_controller_test.go#L42

**Review by camilamacedo86 — APPROVED** [review]
/lgtm

JFY: @NikhilSharmaWe

**Pull request merged**